recipe-server/normandy/health/tests/test_api.py

```diff
@@ -42,39 +50,66 @@ def version_info_dir(self, tmpdir, settings):
     @pytest.fixture
     def commit_file(self, version_info_dir):
         f = version_info_dir.join('commit')
-        f.write('bc685e4be05a182ae819990509c92affa3d882ab\n')
+        f.commit = 'bc685e4be05a182ae819990509c92affa3d882ab'
+        f.write(f.commit + '\n')
         return f
 
     @pytest.fixture
     def tag_file(self, version_info_dir):
         f = version_info_dir.join('tag')
-        f.write('v20\n')
+        f.tag = 'v20'
+        f.write(f.tag + '\n')
         return f
 
-    def test_with_both(self, commit_file, tag_file):
-        commit, tag = views.get_version_info()
-        assert commit == 'bc685e4be05a182ae819990509c92affa3d882ab'
-        assert tag == 'v20'
-
-    def test_with_only_commit(self, commit_file):
-        commit, tag = views.get_version_info()
-        assert commit == 'bc685e4be05a182ae819990509c92affa3d882ab'
-        assert tag == 'unknown'
-
-    def test_with_only_tag(self, tag_file):
-        commit, tag = views.get_version_info()
-        assert commit == 'unknown'
-        assert tag == 'v20'
-
-    def test_with_nothing(self, version_info_dir):
-        commit, tag = views.get_version_info()
-        assert commit == 'unknown'
-        assert tag == 'unknown'
-
-    def test_it_is_cached(self, commit_file, tag_file):
-        commit, tag = views.get_version_info()
-        assert commit == 'bc685e4be05a182ae819990509c92affa3d882ab'
-        assert tag == 'v20'
+    @pytest.fixture
+    def build_time_file(self, version_info_dir):
+        f = version_info_dir.join('build_time')
+        ts = 1493051820
+        f.write('1493051820\n')
+        f.datetime = datetime.utcfromtimestamp(ts)
+        return f
+
+    @pytest.fixture
+    def clear_version_info(self):
```

Comment:
I think this would be better suited to a context manager than a pytest fixture, given that it does cleanup after running some code.